### PR TITLE
Implement canonical urls and remove not needed parameters

### LINKED ISSUE #35133: Canonical Url is missing from the homepage
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

Hi
I've seen that the homepage is without the canonical url. I think it is very useful to prevent duplication issues

using this code into head.tpl the problem is fixed

the better solution should be using Controller::getCanonicalURL() method to assign it to $page.canonical

canonical URLs are used to protect pages from duplication, both internal to the site and external if they made a copy of the site on other servers. at an SEO level, in this way Google would only index the reported page.
you can see it on Google Search Console, for example with sites that have multiple languages and currencies, you would see that Google reports duplication errors because it sees duplications without a canonical URL. A bit like what happens with pagination

bye

### Expected behavior

_No response_

### Steps to reproduce

1. go to the homepage
2. open the source code of your browser
3. search for canonical url

### PrestaShop version(s) where the bug happened

8.1.3

### PHP version(s) where the bug happened

8.1

### If your bug is related to a module, specify its name and its version

_No response_

### Your company or customer's name goes here (if applicable).

_No response_

### PR DESCRIPTION
**Pull request by Hlavtox**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           |  8.1.x
| Description?      | Implements missing canonical urls for all controllers where it makes sense. Removes extra true parameters from getPageLink, as it has no effect.
| Type?             | bug fix
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Tests green
| UI Tests          | https://github.com/Hlavtox/ga.tests.ui.pr/actions/runs/7641255631
| Fixed issue or discussion?     | Fixes #35133
| Related PRs       | 
| Sponsor company   |

### CODE REVIEW

File: controllers/front/ContactController.php [hunk 1]
```diff
@@ -53,4 +53,12 @@ public function getBreadcrumbLinks()
 
         return $breadcrumb;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCanonicalURL()
+    {
+        return $this->context->link->getPageLink('contact', true);
```
Comment (kpodemski) [inline controllers/front/ContactController.php]:
```suggestion
        return $this->context->link->getPageLink('contact');
```

File: controllers/front/IndexController.php [hunk 2]
```diff
@@ -41,4 +41,12 @@ public function initContent()
         ]);
         $this->setTemplate('index');
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCanonicalURL()
+    {
+        return $this->context->link->getPageLink('index', true);
```
Comment (kpodemski) [inline controllers/front/IndexController.php]:
```suggestion
        return $this->context->link->getPageLink('index');
```

File: controllers/front/SitemapController.php [hunk 3]
```diff
@@ -236,4 +236,12 @@ public function getBreadcrumbLinks()
 
         return $breadcrumb;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCanonicalURL()
+    {
+        return $this->context->link->getPageLink('sitemap', true);
```
Comment (kpodemski) [inline controllers/front/SitemapController.php]:
```suggestion
        return $this->context->link->getPageLink('sitemap');
```

File: controllers/front/StoresController.php [hunk 4]
```diff
@@ -230,4 +230,12 @@ public function getBreadcrumbLinks()
 
         return $breadcrumb;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCanonicalURL()
+    {
+        return $this->context->link->getPageLink('stores', true);
```
Comment (kpodemski) [inline controllers/front/StoresController.php]:
```suggestion
        return $this->context->link->getPageLink('stores');
```

**Review by kpodemski — APPROVED** [review]
Since SSL used on half of the pages is no longer the case I see these changes as valid 👍🏻

**Review by M0rgan01 — APPROVED** [review]

**Comment by Hlavtox** [comment]
Test run - https://github.com/Hlavtox/ga.tests.ui.pr/actions/runs/7641255631

**Comment by Hlavtox** [comment]
@florine2623 Tests green, can we merge?

**Comment by florine2623** [comment]
Tests green, OK for product council and for dev, so it can be merged :)
No need QA ^^

**Pull request merged**